### PR TITLE
refactor: update schema to resolve reference property issue

### LINKED ISSUE #165: Parameter defines it's own reference property
#### Describe the bug
Similar to #164 we have a problem with `parameter` defining it's own reference property: https://github.com/asyncapi/spec-json-schemas/blob/a8a5583c2c90ea660243492abcc809fa342cbb86/schemas/2.3.0.json#L554

#### Expected behavior
Parameters should use:
```json
  "additionalProperties": {
    "oneOf": [{"$ref": "#/definitions/parameter"}, {"$ref": "#/definitions/Reference"}]
  },
```

### PR DESCRIPTION
**Pull request by mastdev**
**Description**
- Update the schema for `paramter.json` and `parameters.json`

**Related issue(s)**
Resolves #165

### CODE REVIEW

File: definitions/2.5.0/parameters.json [hunk 1]
```diff
@@ -1,7 +1,11 @@
 {
   "type": "object",
   "additionalProperties": {
-    "$ref": "http://asyncapi.com/definitions/2.5.0/parameter.json"
+    "oneOf": [
+      {
+        "$ref": "http://asyncapi.com/definitions/2.5.0/parameter.json"
+      }
+    ]
```
Comment (derberg) [inline definitions/2.5.0/parameters.json:8]:
```suggestion
    "oneOf": [
      {
        "$ref": "http://asyncapi.com/definitions/2.5.0/Reference.json"
      },
      {
        "$ref": "http://asyncapi.com/definitions/2.5.0/parameter.json"
      }
    ]
```

**Review by derberg — CHANGES_REQUESTED** [review]
@jonaslagoni I'm wondering if it makes sense to backport to previous versions 🤔 it is rather refactor work that an actuall fix

**Comment by derberg** [comment]
@mastdev yo, wanted to check if you saw my review

**Comment by mastdev** [comment]
Sorry, been inactive for a while. Thanks, @derberg for the review. Will incorporate the changes in the following commit.

File: definitions/2.5.0/parameters.json (hunk 1 above)
Comment (mastdev) [reply definitions/2.5.0/parameters.json]: Made the changes with the commit  [#cdee75e](https://github.com/asyncapi/spec-json-schemas/pull/275/commits/cdee75ef79967df52f1a97c3748581e3a5f62679)

**Review by derberg — APPROVED** [review]
lgtm

@jonaslagoni since you reported the issue, I want to make sure you are happy with the outcome

**Review by jonaslagoni — APPROVED** [review]
👍

**Comment by derberg** [comment]
@mastdev thanks a lot for your contribution

there are few more similar issues https://github.com/asyncapi/spec-json-schemas/issues?q=is%3Aissue+is%3Aopen+label%3AHacktoberfest if you would like to help more

**Comment by mastdev** [comment]
Thanks, @derberg for the opportunity. I am new to Open Source and it feels great to have one of my initial PRs merged. 😃 
Sure, I'll take a look at other issues.

**Comment by derberg** [comment]
@mastdev congrats 😄 and thanks for helping us here 🙇🏼

**Pull request merged**